src/System.Drawing.Common/src/System/Drawing/Image.cs

```diff
@@ -99,6 +99,38 @@ void ISerializable.GetObjectData(SerializationInfo si, StreamingContext context)
         /// </summary>
         public static Image FromFile(string filename) => FromFile(filename, false);
 
+        public static Image FromFile(string filename, bool useEmbeddedColorManagement)
+        {
+            if (!File.Exists(filename))
+            {
+                // Throw a more specific exception for invalid paths that are null or empty,
+                // contain invalid characters or are too long.
+                filename = Path.GetFullPath(filename);
+                throw new FileNotFoundException(filename);
+            }
+
+            // GDI+ will read this file multiple times. Get the fully qualified path
+            // so if our app changes default directory we won't get an error
+            filename = Path.GetFullPath(filename);
+
+            IntPtr image = IntPtr.Zero;
+
+            if (useEmbeddedColorManagement)
+            {
+                Gdip.CheckStatus(Gdip.GdipLoadImageFromFileICM(filename, out image));
+            }
+            else
+            {
+                Gdip.CheckStatus(Gdip.GdipLoadImageFromFile(filename, out image));
+            }
+
+            ValidateImage(image);
+
+            Image img = CreateImageObject(image);
```

Comment:
I think in Unix we should be able to cleanup and remove `CreateFromHandle` and just move the code to `CreateImageObject`, no?

Reply:
Those things are rabbit holes 😄. `CreateImageObject` only differed in the exception that is thrown (good, so we streamlined that, too), and the way a `Metafile` object is created from a handle. So I cleaned that up, too. Looks like there are some opportunities left in `Metafile`, let me see if I can cook up a separate PR for that.

Reply:
Cool. Thanks. Taking another look.